libraries/botbuilder-core/botbuilder/core/adapters/test_adapter.py

```diff
@@ -446,6 +447,23 @@ def add_exchangeable_token(
         )
         self.exchangeable_tokens[key.to_key()] = key
 
+    def throw_on_exchange_request(
+        self,
+        connection_name: str,
+        channel_id: str,
+        user_id: str,
+        exchangeable_item: str,
+    ):
+        key = ExchangeableToken(
+            connection_name=connection_name,
+            channel_id=channel_id,
+            user_id=user_id,
+            exchangeable_item=exchangeable_item,
+            token=TestAdapter.__EXCEPTION_EXPECTED,
+        )
+
+        self.exchangeable_tokens[key.to_key()] = key

```

Comment:
I noticed the same code in JavaScript; in JS you can use objects as keys [Maps](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map), so I might change the JS impl to do that (parity with C#). 

I don't recall, but I believe there is something similar in Python...

Note, this isn't actionable for this PR, just potential future code cleanup.